internal/pipe/docker/api_kaniko.go

```diff
@@ -0,0 +1,82 @@
+package docker
+
+import (
+	"fmt"
+	"github.com/goreleaser/goreleaser/pkg/config"
+	"os"
+
+	"github.com/goreleaser/goreleaser/pkg/context"
+)
+
+func init() {
+	registerImager(useKaniko, kanikoImager{})
+}
+
+const (
+	kanikoExecutorImage = "gcr.io/kaniko-project/executor:latest"
+	kanikoWorkspace     = "/workspace"
+)
+
+type kanikoImager struct{}
+
+// Push is a no-op and it returns nil because kaniko performs build and push in a single step.
+// https://github.com/GoogleContainerTools/kaniko#how-does-kaniko-work
+func (i kanikoImager) Push(_ *context.Context, _ string, _ []string) error {
+	return nil
+}
+
+func (i kanikoImager) Build(ctx *context.Context, docker config.Docker, root string, images, flags []string) error {
+	if err := runCommand(ctx, root, "docker", i.buildCommand(docker, images, flags)...); err != nil {
+		return fmt.Errorf("failed to build %s: %w", images[0], err)
+	}
+	return nil
+}
+
+func (i kanikoImager) buildCommand(docker config.Docker, images, flags []string) []string {
+	base := []string{"run"}
+
+	base = i.appendWorkspaceVolumeMount(base)
+	base = i.appendKanikoExectorImage(base)
+	base = i.appendContext(base)
+	base = i.appendDockerfile(base, docker.Dockerfile)
+	base = i.appendDestinations(base, images)
+	base = i.appendNoPush(base, docker.SkipPush == "true")
```

Comment:
my understanding is that it will also push the image on build, correct?

if so, we need to think about another way, users might not want to publish on build, and the skip push option is used to skip the push when running the entire pipeline...

Reply:
Yes by default it pushes on build.